### PR TITLE
Fix Bugs in PP `struct.hpp` (PMACC_C_VECTOR)

### PR DESCRIPTION
**Pull request by psychocoderHPC**
- fix wrong implementation of `PMACC_C_VECTOR`
  - dimension for vector is undefinded
  - type (in this case a vector type) is used instead of
    the type of the components
- fix signature of `PMACC_C_STRING` (takes now only two arguments)
- change interface of `PMACC_PP_INIT_VALUE_VARIABLES`
  - add parameter emptyStruct
- add documentation

### CODE REVIEW

**Comment by psychocoderHPC** [comment]
reassign it to @BenjaminW3 because he review the initial pull request

File: src/libPMacc/include/preprocessor/struct.hpp [hunk 1]
```diff
@@ -256,15 +260,16 @@ namespace namespace_name{                                                      \
...
+        PMACC_PP_INIT_VALUE_VARIABLES(PMACC_PP_X_INIT_VALUE_VARIABLE_WITH_PAREN,((5,(a,b,EmptyStruct2))),__VA_ARGS__) \
+        {}                                                                     \
+                                                                               \
         BOOST_PP_SEQ_FOR_EACH(PMACC_PP_SEQ_MACRO_WITH_ACCESSOR,PMACC_PP_CREATE_C_VALUE_VARIABLE,__VA_ARGS__)  \
         BOOST_PP_SEQ_FOR_EACH(PMACC_PP_SEQ_MACRO_WITH_ACCESSOR,PMACC_PP_CREATE_VALUE_VARIABLE,__VA_ARGS__)    \
         BOOST_PP_SEQ_FOR_EACH(PMACC_PP_SEQ_MACRO_WITH_ACCESSOR,PMACC_PP_CREATE_C_VECTOR_VARIABLE,__VA_ARGS__) \
         BOOST_PP_SEQ_FOR_EACH(PMACC_PP_SEQ_MACRO_WITH_ACCESSOR,PMACC_PP_CREATE_C_STRING_VARIABLE,__VA_ARGS__) \
-	    BOOST_PP_SEQ_FOR_EACH(PMACC_PP_SEQ_MACRO_WITH_ACCESSOR,PMACC_PP_CREATE_EXTENT,__VA_ARGS__)          \
-        BOOST_PP_SEQ_FOR_EACH(PMACC_PP_SEQ_MACRO_WITH_ACCESSOR,PMACC_PP_CREATE_VALUE_VARIABLE_WITH_PAREN,__VA_ARGS__)    \
+	    BOOST_PP_SEQ_FOR_EACH(PMACC_PP_SEQ_MACRO_WITH_ACCESSOR,PMACC_PP_CREATE_EXTENT,__VA_ARGS__)            \
```
Comment (BenjaminW3) [inline src/libPMacc/include/preprocessor/struct.hpp]: Tab...

Comment (ax3l) [inline src/libPMacc/include/preprocessor/struct.hpp]: pfui!

Comment (psychocoderHPC) [inline src/libPMacc/include/preprocessor/struct.hpp]: Test passed :-) well done!

File: src/libPMacc/include/preprocessor/struct.hpp [hunk 2]
```diff
@@ -48,14 +48,18 @@
 
 /** create static const member vector that needs no memory inside of the struct
  *
- *   @param type type of on element
+ *   @param type type of on element (types with comma in are not allowed (e.g. `Vector<type,dim>`)
```
Comment (BenjaminW3) [inline src/libPMacc/include/preprocessor/struct.hpp]: "type of **a**n element"
"types ~~with comma in~~ containing a comma"

**Pull request merged**